man: revamp MPI_File_* and MPI_Register_datarep man pages

ggouaillardet opened the pull request:
- suggest USE mpi instead of INCLUDE 'mpif.h'
- fix indentation
Thanks Jeff for pointing this issue.

(cherry picked from commit open-mpi/ompi@b17c89c1e63d4cc4e9420fc818e0d172156be9c4)

ggouaillardet commented:
:bot:assign: @jsquyres 
:bot:label:enhancement
:bot:milestone:v2.0.0

mellanox-github commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/875/ for details.

jsquyres commented:
Looks like I made comments on open-mpi/ompi@b17c89c1e63d4cc4e9420fc818e0d172156be9c4.  Oops -- sorry about that.  :-)

There are a few blank lines we can eliminate, and one missing `use mpi` notation.

mellanox-github commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/888/ for details.

ggouaillardet commented:
@jsquyres i made the requested changes, squashed them and rebased the PR.
since you commented on the ompi repo, comments were not lost during the rebase :-)

jsquyres commented:
:+1:
